packages/docusaurus/src/commands/build.ts

```diff
@@ -176,15 +176,15 @@ async function buildLocale({
 
     if (configureWebpack) {
       clientConfig = applyConfigureWebpack(
-        configureWebpack.bind(plugin), // The plugin lifecycle may reference `this`. // TODO remove this implicit api: inject in callback instead
+        configureWebpack.bind(plugin), // The plugin lifecycle may reference `this`.
         clientConfig,
         false,
         props.siteConfig.webpack?.jsLoader,
         plugin.content,
       );
 
       serverConfig = applyConfigureWebpack(
-        configureWebpack.bind(plugin), // The plugin lifecycle may reference `this`. // TODO remove this implicit api: inject in callback instead
+        configureWebpack.bind(plugin), // The plugin lifecycle may reference `this`.
```

Comment:
Same argument as above. `this` is actually quite useful, and if plugins return an object, there's no reason why it can't use `this`.

Reply:
I'd rather not. 

Why do we need an implicit this context when things can be provided explicitly as params?

That's also why everybody use React fn components instead of classes nowadays (it's not just because of hooks)

Reply:
The `this` context is not implicit; it's what _should_ happen. If a user returns an object with a property, she should expect to be able to refer to it within the same object. We unbounding the `configureWebpack` method from the `this` context is unintuitive. I agree that it's bad to make it an explicit API and make certain things only accessible on `this`, but we shouldn't deprive people of this API either.